pinax/comments/views.py

```diff
@@ -1,106 +1,134 @@
-import json
-
-from django.http import HttpResponse
-
-from django.views.decorators.http import require_POST
-from django.shortcuts import get_object_or_404, redirect
-from django.template import RequestContext
-from django.template.loader import render_to_string
-
 from django.contrib.auth.decorators import login_required
 from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ImproperlyConfigured
+from django.http import JsonResponse, HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.template import RequestContext
+from django.template.loader import render_to_string
+from django.utils.decorators import method_decorator
+from django.views.generic import CreateView, UpdateView, DeleteView
 
 from .authorization import load_can_delete, load_can_edit
 from .forms import CommentForm
 from .models import Comment
 from .signals import commented, comment_updated
 
-
 can_delete = load_can_delete()
 can_edit = load_can_edit()
 
 
-def dehydrate_comment(comment):
-    return {
-        "pk": comment.pk,
-        "comment": comment.comment,
-        "author": comment.author.username,
-        "name": comment.name,
-        "email": comment.email,
-        "website": comment.website,
-        "submit_date": str(comment.submit_date)
-    }
-
-
-@require_POST
-def post_comment(request, content_type_id, object_id, form_class=CommentForm):
-    content_type = get_object_or_404(ContentType, pk=content_type_id)
-    obj = get_object_or_404(content_type.model_class(), pk=object_id)
-    form = form_class(request.POST, request=request, obj=obj, user=request.user)
-    if form.is_valid():
-        comment = form.save()
-        commented.send(sender=post_comment, comment=comment, request=request)
-        if request.is_ajax():
-            return HttpResponse(json.dumps({
+class CommentSecureRedirectToMixin(object):
+
+    def get_secure_redirect_to(self, object=None):
+        redirect_to = self.request.POST.get("next")
+        # light security check -- make sure redirect_to isn't garbage.
+        if not redirect_to or " " in redirect_to or redirect_to.startswith("http"):
+            try:
+                if object is not None:
+                    url = object.get_absolute_url()
+                elif self.object is not None:
+                    url = self.object.content_object.get_absolute_url()
+            except AttributeError:
+                raise ImproperlyConfigured(
+                    "No URL to redirect to.  Either provide a url or define"
+                    " a get_absolute_url method on the Model.")
+        return redirect_to
+
+
+class CommentCreateView(CommentSecureRedirectToMixin, CreateView):
+    form_class = CommentForm
+    content_object = None
+
+    def get_form_kwargs(self):
+        kwargs = super(CommentCreateView, self).get_form_kwargs()
+        kwargs.update({
+            'request': self.request,
```

Comment:
@arthur-wsw I'm absolutely loving all your contributions so please do not take this as a nit-pick but just wanted to help you round out your contributions to be more in line with [Pinax style standards](http://pinaxproject.com/pinax/how_to_contribute/#coding-style).

We use double-quotes for quoting strings.

Would you mind terribly much in updating this code to use double-quotes?